astro.config.ts

```diff
@@ -60,7 +61,10 @@ export default defineConfig({
 		astroCodeSnippets(),
 	],
 	vite: {
-		plugins: [vitePreact()],
+		ssr: {
+			// Externalize deps of `astro-og-canvas`
+			external: isBuild ? [] : ['canvaskit-wasm', 'entities'],
+		},
```

Comment:
Currently Astro's package crawler only `noExternal` third-party Astro packages, we'd also need to `external` their sub dependencies, and in dev only speficially to avoid `entry.mjs` build errors.

This workarounds the issue for now, but we can implement the feature upstream soon.

Reply:
It would need to be fixed in Astro's [dependency walker](https://github.com/withastro/astro/blob/bf5ad72e81da36c68f96c2431ec547e356538b80/packages/astro/src/core/create-vite.ts#L191) to auto external the packages sub-dependencies too. I'm planning to unify this code so it can be shared between Astro, vite-plugin-svelte and vite-plugin-solid this week, so everything should work OOTB then!